forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -241,3 +278,191 @@ impl WalletCommands {
         }
     }
 }
+
+fn format_balance_string(
+    token_amount: TokenAmount,
+    mode: FormattingMode,
+) -> anyhow::Result<String> {
+    let units = ["atto", "femto", "pico", "nano", "micro", "milli", ""];
+    let num: Decimal = Decimal::try_from_i128_with_scale(
+        token_amount
+            .atto()
+            .to_i128()
+            // currently the amount cannot be more than 2B x 10^18 tokens
+            // the limit here is 20B
+            .ok_or(anyhow::Error::msg("value is too big"))?,
+        0,
+    )?;
+
+    let orig = num;
+
+    let mut num = num;
+    let mut unit_index = 0;
+    while num >= dec!(1000.0) && unit_index < units.len() - 1 {
+        num /= dec!(1000.0);
+        unit_index += 1;
+    }
+
+    let res = match mode {
+        FormattingMode::ExactFixed => {
+            let fil = orig / dec!(1e18);
+            format!("{fil} FIL")
+        }
+        FormattingMode::NotExactFixed => {
+            let fil_orig = orig / dec!(1e18);
+            let fil = fil_orig
+                .round_sf_with_strategy(
+                    NUM_SIGNIFICANT_DIGITS,
+                    RoundingStrategy::MidpointAwayFromZero,
+                )
+                .ok_or(anyhow::Error::msg("cannot represent"))?;
```

Comment:
Is it possible to write a test for this failure case? If not, why not?

Reply:
Looking through the source code of `round_sf_with_strategy` `None` value of `Option` enum is never returned. So basically this is infallible, therefore no tests are needed.

Reply:
It's not exactly infallible but we can leave this for now.